src/mgr/PyModules.cc

```diff
@@ -726,8 +725,9 @@ PyObject* PyModules::get_perf_schema_python(
   } else {
     auto key = DaemonKey(svc_type, svc_id);
     // so that the below can be a loop in all cases
-    if (daemon_state.exists(key)) {
-      states[key] = daemon_state.get(key);
+    auto got = daemon_state.get(key);
+    if (got != nullptr) {
+      daemons[key] = got;
```

Comment:
states[key]

Reply:
Oops, that was a bad backport from a branch where it was renamed.  Fixed.